### PR TITLE
[Feature] XGBoost native multiquantile regression

### PR DESCRIPTION
**Pull request by ozink-u8**
Checklist before merging this PR:
- [x] Mentioned all issues that this PR fixes or addresses.
- [x] Summarized the updates of this PR under **Summary**.
- [x] Added an entry under **Unreleased** in the [Changelog](../CHANGELOG.md).

Fixes #3047 

### Summary

Analogously to #3032 adds multiquantile regression to XGBoost.

#### Change Summary (Claude Code)
- `darts/models/forecasting/xgboost.py`: accept `likelihood="multiquantile"`, set `objective="reg:quantileerror"` and `quantile_alpha=<list>`, update docstrings
- `darts/utils/likelihood_models/sklearn.py`: normalize quantiles to `float` to avoid `XGBoostError` from `np.float64` serialization. It seemed like XGBoost had some for of casting but the sort() would create a list[np.float64] instead of a np.ndarray so their check for a np.ndarray would not catch it.
- `darts/tests/models/forecasting/test_sklearn_models.py`: add `test_model_construction_multiquantile_xgb` and `test_get_estimator_multiquantile_xgb` analogously to #3032 
- `examples/20-SKLearnModel-examples.ipynb`: mention `XGBModel` alongside `CatBoostModel` in the multiquantile tip analogously to #3032 
- `darts/tests/explainability/test_shap_explainer.py`: fix deprecated pandas freq `"d"` → `"D"` (coincidentally also analogously to #3032)

### CODE REVIEW

File: darts/models/forecasting/xgboost.py [hunk 1]
```diff
@@ -239,15 +246,23 @@ def _set_likelihood(
...
         )
 
         if likelihood is None:
             return
 
         if likelihood == LikelihoodType.Poisson.value:
             self.kwargs["objective"] = f"count:{likelihood}"
-        else:  # quantile
+        elif isinstance(self._likelihood, MultiQuantileRegression):
+            # must check before QuantileRegression since MultiQuantileRegression is a subclass
```
Comment (dennisbader) [inline darts/models/forecasting/xgboost.py]: would be cleaner to write it the same way as above, and remove the comment

```suggestion
        elif likelihood == LikelihoodType.MultiQuantile.value:
```

File: darts/models/forecasting/xgboost.py [hunk 2]
```diff
@@ -151,10 +155,13 @@ def encode_year(idx):
...
-            Can be set to `poisson` or `quantile`. If set, the model will be probabilistic, allowing sampling at
-            prediction time. This will overwrite any `objective` parameter.
+            One of ``"multiquantile"``, ``"quantile"``, or ``"poisson"``. If set, the model becomes probabilistic
+            and supports sampling at prediction time. ``"multiquantile"`` and ``"quantile"`` use XGBoost's
+            ``"reg:quantileerror"`` objective; ``"multiquantile"`` fits a single model across all quantiles
+            simultaneously, while ``"quantile"`` fits one model per quantile. This overrides any ``objective``
+            parameter. Default is ``None``.
         quantiles
-            Fit the model to these quantiles if the ``likelihood`` is set to ``"quantile"``.
+            Fit the model to these quantiles if the ``likelihood`` is set to ``"quantile"`` or ``"multiquantile"``.
```
Comment (dennisbader) [inline darts/models/forecasting/xgboost.py:163]: Would be good to use the same description as in CatBoostModel

File: darts/models/forecasting/xgboost.py [hunk 3]
```diff
@@ -151,10 +155,13 @@ def encode_year(idx):
                 To enable past and / or future encodings for any `SKLearnModel`, you must also define the
                 corresponding covariates lags with `lags_past_covariates` and / or `lags_future_covariates`.
         likelihood
-            Can be set to `poisson` or `quantile`. If set, the model will be probabilistic, allowing sampling at
-            prediction time. This will overwrite any `objective` parameter.
+            One of ``"multiquantile"``, ``"quantile"``, or ``"poisson"``. If set, the model becomes probabilistic
+            and supports sampling at prediction time. ``"multiquantile"`` and ``"quantile"`` use XGBoost's
+            ``"reg:quantileerror"`` objective; ``"multiquantile"`` fits a single model across all quantiles
+            simultaneously, while ``"quantile"`` fits one model per quantile. This overrides any ``objective``
+            parameter. Default is ``None``.
```
Comment (dennisbader) [inline darts/models/forecasting/xgboost.py]:
```suggestion
            One of ``"multiquantile"``, ``"quantile"``, or ``"poisson"``. If set, the model becomes probabilistic
            and supports sampling at prediction time. ``"multiquantile"`` and ``"quantile"`` use XGBoost's
            ``"reg:quantileerror"`` objective. Both quantile likelihoods fit dedicated models per quantile, but
            ``"multiquantile"`` can be more efficient due to fewer tabularization operations. This overrides any
            ``objective`` parameter. Default is ``None``.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -10,6 +10,7 @@ but cannot always guarantee backwards compatibility. Changes that may **break co
 ### For users of the library:
 
 - Added native multi-quantile support for `CatBoostModel` by using CatBoost’s `MultiQuantile` loss for faster training and inference. Set `likelihood="multiquantile"` to enable this feature. [#3032](https://github.com/unit8co/darts/pull/3032) by [Zhihao Dai](https://github.com/daidahao)
+- Added native multi-quantile support for `XGBModel` using XGBoost’s `reg:quantileerror` objective. Set `likelihood="multiquantile"` to enable this feature. [#3056](https://github.com/unit8co/darts/pull/3056) by [Oswald Zink](https://github.com/ozink-u8)
```
Comment (dennisbader) [inline CHANGELOG.md]:
```suggestion
- Added native multi-quantile support for `XGBModel`. Similar to the regular quantile support, it still fits dedicated models per quantile, but it is more efficient due to fewer tabularization operations. Set `likelihood="multiquantile"` to enable this feature. [#3056](https://github.com/unit8co/darts/pull/3056) by [Oswald Zink](https://github.com/ozink-u8)
```

File: darts/tests/models/forecasting/test_sklearn_models.py [hunk 5]
```diff
@@ -1862,6 +1862,75 @@ def test_get_estimator_multiquantile(
                 pred_sub_model = sub_model.predict(dummy_feats)[0]
                 np.testing.assert_array_equal(pred_sub_model, pred_j[i])
 
+    @pytest.mark.skipif(not XGB_AVAILABLE, reason="XGBoost required for this test")
```
Comment (dennisbader) [inline darts/tests/models/forecasting/test_sklearn_models.py]: This test is pretty much the same as the one for catboost above. In general we try to avoid code duplication as much as we can :) 

We coud cover both models in the existing test.

You can achieve this adding another level of parametrization to test function decorators:

```
    @pytest.mark.skipif(
        not XGB_AVAILABLE and not CB_AVAILABLE,
        reason="XGBoost or CatBoost required for this test"
    )
    @pytest.mark.parametrize(
        "model_cls,model_kwargs",
        (
            ([(CatBoostModel, cb_test_params)] if CB_AVAILABLE else [])
            + ([(XGBModel, xgb_test_params)] if XGB_AVAILABLE else [])
        )
    )
```

Then you have to slightly adapt the code below to use the new parameters and setup the model dynamically.

File: darts/tests/models/forecasting/test_sklearn_models.py [hunk 6]
```diff
@@ -4315,6 +4384,27 @@ class TestProbabilisticSKLearnModels:
...
+            (
+                XGBModel,
+                {
+                    "lags": 2,
+                    "likelihood": "multiquantile",
+                    "multi_models": True,
+                    **xgb_test_params,
+                },
+                0.4,
+            ),
```
Comment (dennisbader) [inline darts/tests/models/forecasting/test_sklearn_models.py]: I think it's fine to only check the one configuration


```suggestion
```

File: darts/tests/models/forecasting/test_sklearn_models.py [hunk 7]
```diff
@@ -4480,6 +4570,43 @@ def test_model_construction_multiquantile(self):
         assert likelihood.type == LikelihoodType.MultiQuantile
         assert likelihood.quantiles == [0.1, 0.3, 0.5, 0.7, 0.9]
 
+    @pytest.mark.skipif(not XGB_AVAILABLE, reason="XGBoost required for this test")
```
Comment (dennisbader) [inline darts/tests/models/forecasting/test_sklearn_models.py]: the same here as above, let's cover both models in the same test via parametrization

**Review by dennisbader — CHANGES_REQUESTED** [review]
Thanks a lot @ozink-u8, it looks great already :) 

I added some minor comments, after that it's ready to be merged 🚀

**Review by dennisbader — APPROVED** [review]
Looks great now, thanks a lot for the updates @ozink-u8 🚀 

I pushed some minor changes, now everything is good to go 💯
